src/app/components/user-list-main/highlight.directive.ts

```diff
@@ -0,0 +1,18 @@
+import { Directive, ElementRef, Input } from '@angular/core';
+
+@Directive({
+    selector: '[appHighlight]',
+    standalone: true,
+})
+export class HighlightDirective {
+   
+    constructor(
+        private el: ElementRef<HTMLElement>,
+    ) { }
+
+    @Input()
+    set appHighlight(age: number) {
+        this.el.nativeElement.style.backgroundColor = (age >= 18 ? 'red':'white');
```

Comment:
It is better to use CSS classes instead of CSS properties. It is hard extend if operate only with single property (classes can contain several properties). Usually Renderer2 https://angular.dev/api/core/Renderer2 is used (it is safe wrapper which can work on client/server side) to work with HTML elements.

It is also possible to use @HostBinding() decorator to add CSS class to host element. For example:

@HostBinding('class.warning')
get hostClass() {
  return this.age > 18;
}